data-prepper-plugins/obfuscate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/obfuscation/action/OneWayHashActionTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation.action;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+
+public class OneWayHashActionTest implements ObfuscationActionTest {
+
+    private OneWayHashAction oneWayHashAction;
+
+    @Test
+    void testObfuscateWithPatternAsNull() {
+        String message = "Hello";
+        OneWayHashActionConfig config = new OneWayHashActionConfig("SHA-512","",null);
+        String result = new OneWayHashAction(config).obfuscate(message, null,createRecord(message));
+        assertNotNull(result);
+        assertThat(result.length(), equalTo(88));        
```

Comment:
Maybe we should also do a simple validation that the string is not present.

```
assertThat(result, not(containsString("Hello")));
```

Reply:
Added additional assert